src/go/k8s/internal/client/kgo_logging.go

```diff
@@ -0,0 +1,48 @@
+package client
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+const (
+	verboseLevel = 4
+	traceLevel   = 2
+	debugLevel   = 1
+	infoLevel    = 0
+)
+
+// Reference implementation https://github.com/redpanda-data/console/blob/0ba44b236b6ddd7191da015f44a9302fc13665ec/backend/pkg/kafka/config_helper.go#L44
+
+// kgoZapLogger is a franz-go logger adapter for zap.
+type kgoZapLogger struct {
```

Comment:
This is just `kgo-logger.go` but not publicaly exported since it's really only used here in wrapping a controller logger in a `franz-go` compatible interface.